server/src/sim.rs

```diff
@@ -177,7 +181,7 @@ impl Sim {
     }
 }
 
-enum Empty {}
+pub enum Empty {}
```

Comment:
Why was this `pub`ed?

Reply:
When moving ensure_nearby out of the impl block, I made it take a &mut Graph<Empty> as its first parameter because I saw that other code used the Empty type in the template. Then I got the complaint:

```
private type `sim::Empty` in public interface
can't leak private type rustc(E0446)
```

and I made Empty public because I thought that was better than making ensure_nearby not public.

Playing around with the code now, I can see that the function can simply continue to be a template, which is closer to how it originally was, and the enum doesn't need to be made public. Like this: `pub fn ensure_nearby<N>(graph: &mut Graph<N>, start: &Position, distance: f64)` I think I had tried getting that to work earlier but couldn't because I was confused about there being a `Graph` and a `Graph<N>` in two different places in the code. I've made a commit that addresses this point.